### DISCUSSION TITLE
Usage on Android.

### DISCUSSION THREAD

**Discussion by ByteGeek** [JanusGraph/janusgraph#4228] [General]
Has anyone successfully run Janus graph on android?

There are some java related class compatibility issues I am facing and am wondering if this is possible from a technical perspective.

**Comment by hadoopmarc** [comment]
Can you give some examples of the "java related class compatibility issues"? Did you also check https://developer.android.com/studio/write/java11-default-support-table

**Comment by ByteGeek** [comment]
It says invokedyanmic is not supported in API level less than 26.
https://stackoverflow.com/questions/49441260/invokedynamic-on-android-sdk26-oreo

**Reply by hadoopmarc** [reply]
I have no experience with Android (Studio) but API level 26 is from 2017, so I guess you can simply configure your compilation target to not include such old Android versions.

**Comment by ByteGeek** [comment]
Thanks @hadoopmarc 
https://docs.janusgraph.org/operations/deployment/ for the embedded option, could we use[ Cypher for Gremlin ](https://github.com/opencypher/cypher-for-gremlin)to convert  cypher queries to gremlin queries? Are there any examples of folks running embedded Janus graph? Could not find a lot of documentation around it.

**Reply by hadoopmarc** [reply]
The git repository for Cypher for Gremlin does not seem to be maintained, so you should think twice before using it. Embedded Janusgraph is very efficient for bulk loading with Apache Spark, so this is certainly in use. The other major application is building REST API endpoints with a JanusGraph database, e.g. see https://github.com/apache/atlas/blob/master/pom.xml
